charmcraft/cmdbase.py

```diff
@@ -27,6 +31,24 @@ class BaseCommand(craft_cli.BaseCommand):
     - needs_config: will ensure a config is provided when executing the command
 
     The subclass must be declared in the corresponding section of main.COMMAND_GROUPS.
+
+    If the command may produce the result in a programmatic-friendly format, it
+    should call the 'include_format_option' method to properly affect the parser and
+    then emit only one message with the result of the 'format_content' method.
     """
 
     needs_config = False
+
+    def format_content(self, format_, content):
```

Comment:
Personally I would prefer fmt to format_ :-)

Reply:
good idea